src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1474,7 +1476,7 @@ private MethodSymbol FindEntryPoint(CancellationToken cancellationToken, out Imm
                     if (candidate.IsAsync)
                     {
                         // PROTOTYPE(async-main): Get the diagnostic to point to a smaller syntax piece.
-                        CheckFeatureAvailability(candidate.DeclaringSyntaxReferences.Single().GetSyntax(), MessageID.IDS_FeatureAsyncMain, diagnostics);
+                        asyncOk &= CheckFeatureAvailability(candidate.DeclaringSyntaxReferences.Single().GetSyntax(), MessageID.IDS_FeatureAsyncMain, possibleAsyncMainDiagnostics);
```

Comment:
>asyncOk &= CheckFeatureAvailability(candidate.DeclaringSyntaxReferences.Single().GetSyntax(), MessageID.IDS_FeatureAsyncMain, possibleAsyncMainDiagnostics) [](start = 24, length = 155)

It feels like if the feature is not available we should add a warning into ```warnings``` and skip the candidate.